hooks/helpers/useOnChange.ts

```
@@ -0,0 +1,42 @@
+import { useErrorContext } from "hooks";
+import { ChangeEvent } from "react";
+
+export function useOnChange(
+  onInput: (value: string, ...args: unknown[]) => void,
+  maxDecimals: number,
+  allowNegative: boolean,
+  type = "number"
+) {
+  const { addErrorMessage, removeErrorMessage } = useErrorContext();
```

handles showing and hiding error for too many decimals